app/src/main/java/org/schabi/newpipe/fragments/detail/DescriptionFragment.java

```diff
@@ -57,37 +71,196 @@ public void onDestroy() {
-                        description.getContent(), descriptionTextView);
+                        description.getContent(), binding.detailDescriptionView);
                 break;
         }
     }
+
+
+    private void setupMetadata(final LayoutInflater inflater,
+                               final LinearLayout layout) {
+        addMetadataItem(inflater, layout, false,
+                R.string.metadata_category, streamInfo.getCategory());
+
+        addTagsMetadataItem(inflater, layout);
+
+        addMetadataItem(inflater, layout, false,
+                R.string.metadata_licence, streamInfo.getLicence());
+
+        addPrivacyMetadataItem(inflater, layout);
+
+        if (streamInfo.getAgeLimit() != NO_AGE_LIMIT) {
+            addMetadataItem(inflater, layout, false,
+                    R.string.metadata_age_limit, String.valueOf(streamInfo.getAgeLimit()));
+        }
+
+        if (streamInfo.getLanguageInfo() != null) {
+            addMetadataItem(inflater, layout, false,
+                    R.string.metadata_language, streamInfo.getLanguageInfo().getDisplayLanguage());
+        }
+
+        addMetadataItem(inflater, layout, true,
+                R.string.metadata_support, streamInfo.getSupportInfo());
+        addMetadataItem(inflater, layout, true,
+                R.string.metadata_host, streamInfo.getHost());
+        addMetadataItem(inflater, layout, true,
+                R.string.metadata_thumbnail_url, streamInfo.getThumbnailUrl());
+    }
+
+    private void addMetadataItem(final LayoutInflater inflater,
+                                 final LinearLayout layout,
+                                 final boolean linkifyContent,
+                                 @StringRes final int type,
+                                 @Nullable final String content) {
+        if (isBlank(content)) {
+            return;
+        }
+
+        final ItemMetadataBinding itemBinding
+                = ItemMetadataBinding.inflate(inflater, layout, false);
+
+        itemBinding.metadataTypeView.setText(type);
+        itemBinding.metadataTypeView.setOnLongClickListener(v -> {
+            ShareUtils.copyToClipboard(requireContext(), content);
+            return true;
+        });
+
+        if (linkifyContent) {
+            TextLinkifier.createLinksFromPlainText(requireContext(),
+                    content, itemBinding.metadataContentView);
+        } else {
+            itemBinding.metadataContentView.setText(content);
+        }
+
+        layout.addView(itemBinding.getRoot());
+    }
+
+    private void addTagsMetadataItem(final LayoutInflater inflater, final LinearLayout layout) {
+        if (streamInfo.getTags() != null && !streamInfo.getTags().isEmpty()) {
+            final ItemMetadataTagsBinding itemBinding
+                    = ItemMetadataTagsBinding.inflate(inflater, layout, false);
+
+            Collections.sort(streamInfo.getTags());
```

Comment:
I'm not confident about this, but maybe the tags should be copied before being mutated?

Reply:
Done